<?xml version="1.0"?>

<launch>
  <arg name="namespace" default="/"/>
  <arg name="cmd_timeout" default="0.5"/>
  <arg name="world_name" default="Empty_world"/>

  <!-- Vehicle pose -->
  <arg name="x" default="0.0"/>
  <arg name="y" default="0.0"/>
  <arg name="z" default="0.0"/>
  <arg name="roll" default="0.0"/>
  <arg name="pitch" default="0.0"/>
  <arg name="yaw" default="0"/>

  <include file="$(find ackermann_vehicle_description)/launch/ackermann_vehicle.launch">
    <arg name="namespace" value="$(arg namespace)"/>
  </include>

  <group ns="$(arg namespace)">

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(arg world_name)"/>
    </include>

    <node name="spawn_vehicle" pkg="gazebo_ros" type="spawn_model"
          args="-urdf -param robot_description -model ackermann_vehicle
                -gazebo_namespace /$(arg namespace)/gazebo
                -x $(arg x) -y $(arg y) -z $(arg z)
                -R $(arg roll) -P $(arg pitch) -Y $(arg yaw)"/>

    <node name="controller_spawner" pkg="controller_manager" type="spawner"
          args="$(find ackermann_vehicle_gazebo)/config/em_3905_joint_ctrlr_params.yaml"/>

    <node name="ackermann_controller" pkg="ackermann_vehicle_gazebo"
          type="ackermann_controller">
      <param name="cmd_timeout" value="$(arg cmd_timeout)"/>
      <rosparam file="$(find ackermann_vehicle_gazebo)/config/em_3905_ackermann_ctrlr_params.yaml" command="load"/>
    </node>

    <!--node name="teleop_ackermann" pkg="ackermann_vehicle_navigation" type = "teleop_ackermann.py" output="screen" launch-prefix="gnome-terminal command"/>-->
    
    <node pkg="ackermann_vehicle_navigation" name="planner" type="planner.py" output="screen">
      <param name="x_desired" value="13.0"/>
      <param name="max_speed" value="1.0"/>
      <param name="min_speed" value="0.5"/>
      <param name="lane_1_desired" value="0.0"/>
      <param name="lane_2_desired" value="1.0"/>
    </node>

    <node pkg="ackermann_vehicle_navigation" name="Autonomous_Systems_MS_3_CLR_Alg_2_Lateral_Team_21" type="Autonomous_Systems_MS_3_CLR_Alg_2_Lateral_Team_21.py" output="screen">
      <param name="Wheel_base" value="0.4239"/>
    </node>
  
    <node pkg="ackermann_vehicle_navigation" name="Autonomous_Systems_MS_3_CLR_Alg_1_Speed_Team_21" type="Autonomous_Systems_MS_3_CLR_Alg_1_Speed_Team_21.py" output="screen">
    </node>
    
    <node pkg="ackermann_vehicle_navigation" name="ackermann_cmd_data_fuser" type="ackermann_cmd_data_fuser.py" output="screen">
    </node>

    <node pkg="ackermann_vehicle_navigation" name="ekf" type="ekf.py" output="screen">
    </node>

    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--standalone rqt_graph" output="screen"/> 

  </group>
</launch>